eng/Version.Details.xml

```diff
@@ -23,6 +23,10 @@
       <Uri>https://github.com/dotnet/roslyn</Uri>
       <Sha>3fc97de0f151e67a0b8ee2840edb04b49fbc623f</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler" Version="4.6.0-3.23164.6">
+      <Uri>https://github.com/dotnet/roslyn</Uri>
+      <Sha>3fc97de0f151e67a0b8ee2840edb04b49fbc623f</Sha>
+    </Dependency>
```

Comment:
Not sure how to make source build succeed. The error is:

```
.packages/microsoft.dotnet.arcade.sdk/8.0.0-beta.23224.3/tools/SourceBuild/AfterSourceBuild.proj(68,5): error : (NETCORE_ENGINEERING_TELEMETRY=AfterSourceBuild) 1 new pre-builts discovered! Detailed usage report can be found at /__w/1/s/artifacts/source-build/self/prebuilt-report/baseline-comparison.xml.
See aka.ms/dotnet/prebuilts for guidance on what pre-builts are and how to eliminate them.
Package IDs are:
Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.4.6.0-3.23164.6
```

This is normal Roslyn project that should be available in source build https://github.com/dotnet/roslyn/blob/main/src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj

@dotnet/source-build-internal what do I do here? thanks

Reply:
Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler does not appear to be built as part of source-build.  You are adding a reference to it from a component that is part of source-build which is therefore introducing a prebuilt.  To resolve this, Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler would need to be included in source-build including any of it's dependencies.

Roslyn tools are excluded from source-build by default by [this logic](https://github.com/dotnet/roslyn/blob/main/src/Tools/Directory.Build.props).

cc @mmitche

Reply:
Thanks @MichaelSimons, but how is it possible that for example Microsoft.CodeAnalysis.ExternalAccess.Razor (*not .RazorCompiler*) referenced just above is fine even though it should be excluded from source build by the same logic (I don't see `<ExcludeFromSourceBuild>false</ExcludeFromSourceBuild>` in its project file https://github.com/dotnet/roslyn/blob/main/src/Tools/ExternalAccess/Razor/Microsoft.CodeAnalysis.ExternalAccess.Razor.csproj).

Reply:
Adding a ProductDependency in the versions.details.xml on a non-source-built component is not an issue unless it is referenced by a source-built component.  The issue with Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler is that it is referenced by Microsoft.NET.Sdk.Razor.SourceGenerators which is included in source-build.

You asked about Microsoft.CodeAnalysis.ExternalAccess.Razor which is also not part of source-build.  My cursory search shows it is referenced by two project both of which are excluded from source-build (Microsoft.CodeAnalysis.Razor.Workspaces.csproj and Microsoft.VisualStudio.RazorExtension.Dependencies.csproj).